cli/cli_app.go

```diff
@@ -278,6 +282,34 @@ var TERRAFORM_HELP_FLAGS = []string{
 	"-h",
 }
 
+// Returns a map of macro placeholders to a function that returns what the placeholder should be replaced with
+func getMacroReplacements(terragruntOptions *options.TerragruntOptions) map[string]func() string {
+	return map[string]func() string{
+		"::TERRAGRUNT_DIR::": func() string { return filepath.Dir(terragruntOptions.TerragruntConfigPath) },
+	}
+}
+
+// Takes terragruntOptions and mutates the args list such that all macros are replaced with their substitutions,
+// if SubstituteMacros is true.
+func performMacroSubstitutions(terragruntOptions *options.TerragruntOptions) error {
+	if terragruntOptions.SubstituteMacros {
+		newTerraformCliArgs := []string{}
+		for _, arg := range terragruntOptions.TerraformCliArgs {
+			newArg := arg
+			for macro, replacementFunc := range getMacroReplacements(terragruntOptions) {
+				if strings.Contains(arg, macro) {
+					newArg = strings.ReplaceAll(arg, macro, replacementFunc())
```

Comment:
Can happen situation when `newArg` will contain another macro, and it will not be replaced?